components/scream/src/physics/shoc/shoc_update_prognostics_implicit_impl.hpp

```diff
@@ -57,10 +63,21 @@ void Functions<S,D>::update_prognostics_implicit(
   auto dl = Kokkos::subview(dl_workspace, Kokkos::make_pair(0,nlev));
   auto d  = Kokkos::subview(d_workspace,  Kokkos::make_pair(0,nlev));
 
-  const auto last_nlev_pack = (nlev-1)/Spack::n;
-  const auto last_nlev_indx = (nlev-1)%Spack::n;
-  const auto last_nlevi_pack = (nlevi-1)/Spack::n;
-  const auto last_nlevi_indx = (nlevi-1)%Spack::n;
+  // 2d allocations for solver RHS
+  const int num_wind_transpose_packs = ekat::npack<Spack>(2);
+  const int num_qtracers_transpose_packs = ekat::npack<Spack>(num_qtracers+3);
+
+  const int n_wind_slots = num_wind_transpose_packs*Spack::n;
+  const int n_trac_slots = num_qtracers_transpose_packs*Spack::n;
+
+  const auto wind_slot    = workspace.template take_macro_block<Scalar>("wind_slot",n_wind_slots);
+  const auto tracers_slot = workspace.template take_macro_block<Scalar>("tracers_slot",n_trac_slots);
+
+  // Reshape 2d views
+  const auto wind_rhs     = uview_2d<Spack>(reinterpret_cast<Spack*>(wind_slot.data()),
+                                            nlev, num_wind_transpose_packs);
+  const auto qtracers_rhs  = uview_2d<Spack>(reinterpret_cast<Spack*>(tracers_slot.data()),
+                                            nlev, num_qtracers_transpose_packs);
```

Comment:
I reserve `num_packs*PackSize` contiguous, Scalar slots, then recast those into a 2d view of size `(nlev, num_packs)`.